test/workbox-background-sync/sw/test-Queue.mjs

```diff
@@ -22,52 +19,51 @@ const getObjectStoreEntries = async () => {
 };
 
 const createSyncEventStub = (tag) => {
+  const extendLifetimePromises = [];
+
   const ret = {
     event: new SyncEvent('sync', {tag}),
     // Default to resolving in the next microtask.
-    done: Promise.resolve(),
-
+    get done() {
+      return Promise.all(extendLifetimePromises);
+    },
   };
 
   // Browsers will throw if code tries to call `waitUntil()` on a user-created
   // sync event, so we have to stub it.
   ret.event.waitUntil = (promise) => {
     // If `waitUntil` is called, defer `done` until after it resolves.
     if (promise) {
-      // Catch failures since all we care about is finished.
-      ret.done = promise.catch(() => undefined).then(ret.done);
+      extendLifetimePromises.push(promise.catch((e) => e));
     }
   };
 
   return ret;
 };
 
-// TODO(philipwalton): uncomment once we move away from the IDB mocks.
-// const clearIndexedDBEntries = async () => {
-//   // Open a conection to the database (at whatever version exists) and
-//   // clear out all object stores. This strategy is used because deleting
-//   // databases inside service worker is flaky in FF and Safari.
-//   // TODO(philipwalton): the version is not needed in real browsers, so it
-//   // can be removed when we move away from running tests in node.
-//   const db = await new DBWrapper('workbox-background-sync').open();
-
-//   // Edge cannot convert a DOMStringList to an array via `[...list]`.
-//   for (const store of Array.from(db.db.objectStoreNames)) {
-//     await db.clear(store);
-//   }
-//   await db.close();
-// };
+const clearIndexedDBEntries = async () => {
+  // Open a conection to the database (at whatever version exists) and
+  // clear out all object stores. This strategy is used because deleting
+  // databases inside service worker is flaky in FF and Safari.
+  // TODO(philipwalton): the version is not needed in real browsers, so it
```

Comment:
That TODO suggests that we could remove this code, right? Although this whole block was itself commented out... so I'm confused.

Reply:
At some point I had the DB version hard-coded in there (which was required by the mocking library). Since that's not there anymore I think I can just remove the TODO comment.